packages/core/services/signers/SignerFactory.js

```diff
@@ -0,0 +1,156 @@
+  const account = getPersistAccount(state);
+  const targetDevice = account.decryptedWallet.entry.encrypted[0].type;
+  devices[WALLET_TYPE_METAMASK] = MetamaskPlugin;
+
+  if(devices.hasOwnProperty(targetDevice)) {
+    const networkData = getSelectedNetwork()(state);
+    const network = networkList[networkData[blockchainType]];
+    const deviceConfig = { network };
+
+    if(targetDevice === WALLET_TYPE_MEMORY) {
+      deviceConfig[privateKeyFieldName] = account.decryptedWallet.privateKey.slice(2, 66);
+    }
+
+    return targetDevice === WALLET_TYPE_METAMASK ? new MetamaskPlugin() : new devices[targetDevice](deviceConfig);
+  }
+};
+
+export const getSigner = (blockchainType, state) => {
+  switch (blockchainType) {
+    case BLOCKCHAIN_BITCOIN: return getBitcoinSigner(state);
+    case BLOCKCHAIN_BITCOIN_CASH: return getBitcoinCashSigner(state);
+    case BLOCKCHAIN_DASH: return getDashSigner(state);
+    case BLOCKCHAIN_LITECOIN: return getLitecoinSigner(state);
+    case BLOCKCHAIN_ETHEREUM: return getEthereumSigner(state);
+    case BLOCKCHAIN_NEM: return getNemSigner(state);
+    case BLOCKCHAIN_WAVES: return getWavesSigner(state);
+  }
+};
+
+export const getBitcoinSigner = (state) => {
+  return getConfiguredSigner (BLOCKCHAIN_BITCOIN, state, {
+    [WALLET_TYPE_MEMORY]: BitcoinMemoryDevice,
+    [WALLET_TYPE_LEDGER]: BitcoinLedgerDeviceMock,
+    [WALLET_TYPE_LEDGER_MOCK]: BitcoinLedgerDeviceMock,
+    [WALLET_TYPE_TREZOR]: BitcoinTrezorDeviceMock,
+    [WALLET_TYPE_TREZOR_MOCK]: BitcoinTrezorDeviceMock
+  }, bitcoin.networks);
+};
+
+export const getBitcoinCashSigner = (state) => {
+  return getConfiguredSigner (BLOCKCHAIN_BITCOIN_CASH, state, {
+    [WALLET_TYPE_MEMORY]: BitcoinCashMemoryDevice,
+    [WALLET_TYPE_LEDGER]: BitcoinCashLedgerDeviceMock,
+    [WALLET_TYPE_LEDGER_MOCK]: BitcoinCashLedgerDeviceMock,
+    [WALLET_TYPE_TREZOR]: BitcoinCashTrezorDeviceMock,
+    [WALLET_TYPE_TREZOR_MOCK]: BitcoinCashTrezorDeviceMock
+  }, bitcoin.networks);
+};
+
+export const getDashSigner = (state) => {
+  return getConfiguredSigner (BLOCKCHAIN_DASH, state, {
+    [WALLET_TYPE_MEMORY]: DashMemoryDevice
+  }, bitcoin.networks);
+};
+
+export const getLitecoinSigner = (state) => {
+  return getConfiguredSigner (BLOCKCHAIN_LITECOIN, state, {
+    [WALLET_TYPE_MEMORY]: BitcoinMemoryDevice,
+    [WALLET_TYPE_LEDGER]: BitcoinLedgerDeviceMock,
+    [WALLET_TYPE_LEDGER_MOCK]: BitcoinLedgerDeviceMock,
+    [WALLET_TYPE_TREZOR]: BitcoinTrezorDeviceMock,
+    [WALLET_TYPE_TREZOR_MOCK]: BitcoinTrezorDeviceMock
+  }, bitcoin.networks);
+};
+
+export const getEthereumSigner = (state) => {
+  const account = getPersistAccount(state);
+
+  switch (account.selectedWallet.type) {
+    case WALLET_TYPE_MEMORY: {
+      return new EthereumMemoryDevice(account.decryptedWallet.privateKey);
+    }
+    case WALLET_TYPE_LEDGER: {
+      return new EthereumTrezorDeviceMock();
```

Comment:
EthereumLedgerDevice

Reply:
I just copied it as it was. I'll fix it.

Reply:
By some reason you left import of EthereumLedgerDevice commented (see https://github.com/ChronoBank/ChronoMint/blob/develop/packages/core/redux/persistAccount/selectors.js#L20). Once I added the import rebuild time increased dramatically.